pkg/crd/schema.go

```diff
@@ -304,6 +304,7 @@ func mapToSchema(ctx *schemaContext, mapType *ast.MapType) *apiext.JSONSchemaPro
 			return &apiext.JSONSchemaProps{}
 		}
 	case *ast.StarExpr:
+		ctx.pkg.AddWarning(loader.ErrFromNode(fmt.Errorf("map values should be a named type, not %T", mapType.Value), mapType.Value))
```

Comment:
maybe tweak the message here a bit so it's obvious it's a warning.  I'd do something like

```
as per the Kubernetes API conventions, map values *should* generally be primitives or named types that alias primitives, not %T.  Consider using the "list as map" pattern here, except in cases where map is used for compatiblity/legacy purposes.
```